File: SearchView/app/src/main/java/mondal/tech/MyRecyclerViewAdapter.java
package mondal.tech;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

public class MyRecyclerViewAdapter extends RecyclerView.Adapter<MyRecyclerViewAdapter.MyViewHolder> {

    private MainActivity mainActivity;
    private ArrayList<String> countryName;

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i)
    {
        View view=LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.single_row_apperance,viewGroup,false);
        MyViewHolder viewHolder=new MyViewHolder(view);
        return viewHolder;
    }

    public void setDataSource(ArrayList<String> countryName)
    {
        this.countryName=countryName;
    }

    public MyRecyclerViewAdapter(MainActivity mainActivity, ArrayList<String> countryName)
    {
        this.mainActivity=mainActivity;
        this.countryName=countryName;
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder viewHolder, int i)
    {

        viewHolder.textView.setText(countryName.get(i));

    }

    @Override
    public int getItemCount() {
        return countryName.size();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder
    {
        public TextView textView;

        public MyViewHolder(@NonNull View itemView)
        {
            super(itemView);
            textView=itemView.findViewById(R.id.text_view);

        }
    }

}

File: KotlinDemoApp3/app/src/main/java/com/nisith/kotlindemoapp3/Adapters/MyRecyclerViewAdapter.kt
package com.nisith.kotlindemoapp3.Adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.nisith.kotlindemoapp3.Model.Country
import com.nisith.kotlindemoapp3.R

class MyRecyclerViewAdapter(var countryList: ArrayList<Country>, var itemClickListener: OnItemClickListener): RecyclerView.Adapter<MyRecyclerViewAdapter.MyViewHolder>() {

    interface OnItemClickListener{
        fun onItemClick(countryName: String)
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view: View = LayoutInflater.from(parent.context).inflate(R.layout.single_row_appearence, parent, false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.countryNameTextView.text = countryList[position].countryName
    }


    override fun getItemCount(): Int {
        return countryList.size
    }


    inner class MyViewHolder(itemView: View): RecyclerView.ViewHolder(itemView) {
        var countryNameTextView: TextView = itemView.findViewById(R.id.country_name_text_view)
        init {
            itemView.setOnClickListener(View.OnClickListener {
                itemClickListener.onItemClick(countryList[absoluteAdapterPosition].countryName)
            })
        }
    }

}